src/sagas/publicChannels/publicChannels.selectors.ts

```diff
@@ -130,12 +130,102 @@ export const currentChannelDisplayableMessages = createSelector(
         id: message.id,
         type: message.type,
         message: message.message,
-        createdAt: formatMessageDisplayDate(message.createdAt),
+        createdAt: message.createdAt,
         nickname: user.username,
       };
     })
 );
 
+// returns array of messages SORTED by createdAt
+export const currentChannelMessagesSortedByTime = createSelector(
+  currentChannelDisplayableMessages,
+  (messages) =>
+    messages.sort((a, b) => b.createdAt - a.createdAt)
+);
+
+// returns array of messages SORTED and GROUPED by createdAt and user
+export const currentChannelMessagesGroupedByUserTime = createSelector(
+  currentChannelMessagesSortedByTime,
+  (messages) => {
+    const timeOfGroupingMessages = 300
+    let newMessages = []
+
+    for (let indexOfMessages = 0; indexOfMessages < messages.length; indexOfMessages++) {
```

Comment:
Are you sure it'll work if there's more than 2 messages that match the condition?
 I mean, you iterate on EACH message and you check if any of the messages that comes next fits the requirements. So with three or more messages I suppose you'll bind them together with the pattern 
`{ message1: "message1 + message2 + message3" }`
then again 
`{ message2: "message2 + message3"}`

Reply:
Maybe it'd be more reliable if you used `reduce` method instead of `for` loop?

Reply:
for sure it works, i increasing indexOfMessages when message will group and for loop have same indexOfMessages and it will take next message
`message4` after `message1 + message2 + message3`